fiftyone/core/labels.py

```
@@ -1193,23 +1200,30 @@ def import_mask(self, update=False):
-        if self.mask_path is not None:
+        if self.mask_path is not None and not overwrite_path:
             etau.copy_file(self.mask_path, outpath)
-        else:
+        elif self.mask is not None:
             _write_mask(self.mask, outpath)
 
             if update:
                 self.mask = None
                 self.mask_path = outpath
+        else:
+            raise ValueError("Detection has no mask or mask_path to export")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)